### ISSUE TITLE
[Bug]: Various commands crash on entry if Discord client is not installed

### ISSUE THREAD

**Issue by 00dani** [chrisgrieser/shimmering-obsidian#52]
### Bug Description

Shimmering Obsidian has a number of commands that try to copy "Discord-ready links" wrapped in `<>` if Discord is focused, such as the `of` forum category selector and the `op` plugin search. Unfortunately, if you don't happen to have the Discord.app client installed on your Mac, those commands simply fail to work, rather than running normally but without special Discord-ready support.

I found this because I have Discord PTB installed instead, which of course is an application with a different name. It's probably unlikely that Obsidian users would have no Discord client at all, since the Obsidian Members Group is so helpful, but many of us are Power Users™ so it's reasonably likely we might have beta releases of Discord installed.

I've done some fiddling with the underlying scripts and found that this line is the culprit:
```javascript
const discordReadyLinks = Application("Discord").frontmost();
```
There's practically no documentation for JXA provided by Apple, so I couldn't find out whether there's an intended "safe" way to call `Application('Discord')` that won't throw an error when it doesn't exist. I don't think Apple like JXA very much.

However, protecting the call to `Application` with a simple `try` block does work, so doing something like this should work fine. (I tried it in Script Editor. It does indeed work fine.)
```javascript
function SafeApplication(appId) {
  try {
    return Application(appId);
  } catch (e) {
    return null;
  }
}

const discordReadyLinks = ['Discord', 'Discord PTB', 'Discord Canary'].some(app => SafeApplication(app)?.frontmost());
```

### Debugging Log

```text
[20:14:09.346] Shimmering Obsidian[Script Filter] Queuing argument '(null)'
[20:14:09.511] Shimmering Obsidian[Script Filter] Script with argv '(null)' finished
[20:14:09.526] ERROR: Shimmering Obsidian[Script Filter] Code 1: /Users/dani/Dropbox (Personal)/Apps/Alfred/Alfred.alfredpreferences/workflows/user.workflow.EED411DC-9859-46C4-9687-004751751E2B/scripts/forum-search.js: execution error: Error: Error: Application can't be found. (-2700)
```

### Workflow Configuration

I don't think these options are super relevant! But here they are anyways:
<img width="495" alt="Screen Shot 2022-01-27 at 20 14 43" src="https://user-images.githubusercontent.com/610876/151328447-4e359fef-eb58-4a03-af1b-6463641c0b3d.png">

### Version & System Information

```text
-------------------------------
Metadata JSONs: 3/3
-------------------------------
INSTALLED VERSION
macOS: 12.2
Obsidian: 0.13.19
Alfred: 4.6.1
Workflow: 2.19.1
Advanced URI Plugin: 1.17.0
Metadata Extractor: 0.3.1
-------------------------------
LATEST VERSION
Obsidian: 0.13.19
Obsidian (Insider): 0.13.22
Workflow: 2.19.1
Advanced URI Plugin: 1.17.0
Metadata Extractor: 0.3.1
-------------------------------
```

### Make sure you have done the following

- [X] I have [updated to the latest version](https://github.com/chrisgrieser/shimmering-obsidian/releases/latest) of this workflow.
- [X] I have read the [Troubleshooting Documentation](https://github.com/chrisgrieser/shimmering-obsidian/blob/main/documentation/Troubleshooting.md).
- [X] If the `o ...` main search does not work, I have tried rerunning `osetup` and it still does not work.

**Comment by chrisgrieser** [comment]
Ah good find, I haven't thought of the possibility of users not having Discord installed. 😅 

And also thanks for looking into a solution, `try catch` is also what I would have tried. 

I am not at my desk rn, but I'll implement it later. However, your suggestion looks good to me as is, so you can also make a PR if you want.

**Comment by chrisgrieser** [comment]
fixed with release 2.19.2 !
